core/models/department.py

```diff
@@ -0,0 +1,27 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+
+
+class Department(models.Model):
+    """ Stores all departments """
+
+    name = models.CharField(unique=True, max_length=50)
+    created_at = models.DateTimeField(auto_now_add=True, editable=False)
+    last_modified = models.DateTimeField(auto_now=True, editable=False)
+
+    def clean(self):
+        self.name = " ".join(self.name.title().split())
+
+    def save(self, *args, **kwargs):
+        try:
+            self.full_clean()
+        except Exception as e:
+            raise ValidationError(e)
+        super().save(*args, **kwargs)
+
+    class Meta:
+        verbose_name = "Department"
+        unique_together = ("name", )
```

Comment:
Having one field for [`unique_together`](https://docs.djangoproject.com/en/2.0/ref/models/options/#unique-together) defeats the purpose. See docs.